runtime/major_gc.c

```diff
@@ -664,21 +665,22 @@ static void update_major_slice_work(intnat howmuch,
                                     int may_access_gc_phase)
 {
   double heap_words;
-  intnat alloc_work, dependent_work, extra_work, new_work;
-  intnat my_alloc_count, my_alloc_direct_count, my_dependent_count;
+  intnat alloc_work, extra_work, new_work;
+  intnat my_alloc_count, my_alloc_direct_count;
```

Comment:
I'm only skimming this function as I review this PR because I know it's about to change radically (in the next week or so).

Reply:
The only claim I'm making here is that I haven't changed the policy, at least in the steady-state case.

Previously, `dependent_allocated` was always zero and `extra_work` went from `0` to `1`, growing by `s / get_custom_max_major()` on off-heap allocation of size `s`. Now instead `allocated_dependent_bytes` is an absolute byte count, and the division by `get_custom_max_major()` happens here instead.

(In the non-steady-state case there's a slight difference: now we're sampling `get_custom_max_major()` at slice time rather than at allocation time, at which point the heap size may be slightly different. I don't think this makes any difference in practice)